pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -112,28 +114,60 @@ func (d *DualWriterMode2) List(ctx context.Context, options *metainternalversion
 	return ll, nil
 }
 
-func enrichObject(orig, copy runtime.Object) (runtime.Object, error) {
-	accessorC, err := meta.Accessor(copy)
-	if err != nil {
-		return nil, err
-	}
-	accessorO, err := meta.Accessor(orig)
-	if err != nil {
-		return nil, err
-	}
-
+func enrichObject(accessorO, accessorC metav1.Object) {
 	accessorC.SetLabels(accessorO.GetLabels())
 
 	ac := accessorC.GetAnnotations()
 	for k, v := range accessorO.GetAnnotations() {
 		ac[k] = v
 	}
 	accessorC.SetAnnotations(ac)
+}
+
+// Update overrides the behavior of the generic DualWriter and writes first to LegacyStorage and then to Storage.
+func (d *DualWriterMode2) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+	legacy, ok := d.Legacy.(rest.Updater)
+	if !ok {
+		return nil, false, errNoUpdateMethod
+	}
+
+	// get old and new (updated) object so they can be stored in legacy store
+	old, err := d.Storage.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		klog.FromContext(ctx).Error(err, "object not found in US for update", "mode", Mode2)
+		return nil, false, err
+	}
```

Comment:
in this case we should convert the storage operation to a create rather than returning an error

Reply:
old will be the object from legacy storage in this case

Reply:
> in this case we should convert the storage operation to a create rather than returning an error

Creating an object if not found in Storage should be handled as part of the update process by [GuaranteedUpdate](https://github.com/grafana/grafana/blob/main/pkg/services/apiserver/storage/entity/storage.go#L513) as described in [this PR](https://github.com/grafana/grafana/pull/85206) though or am I mistaken?

Reply:
I got mixed up. We don't currently have a way of setting `IgnoreNotFound` so we _do_ need to create the object.